src/lib/openapi/spec/constraint-schema.test.ts

```diff
@@ -0,0 +1,34 @@
+import { validateSchema } from '../validate';
+import { ConstraintSchema } from './constraint-schema';
+
+test('constraintSchema', () => {
```

Comment:
I don't really understand these tests. Could you enlighten me?

Reply:
Yeah, they're a bit too clever. This assignment ensures that the generated type works as expected:

```
data: ConstraintSchema = ...
```

(The generated types often turn into `never` if we forget something, which would make this test fail.)

The other tests check that the validation works as expected by testing snapshots of errors for various invalid data.